### PR TITLE
[#1501] Add tests for feedback text question

### PR DESCRIPTION
**Pull request by hhdqirui**
Part of #1501 

**Outline of Solution**
Create tests for existing methods that are not tested.

### CODE REVIEW

File: src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java [hunk 1]
```diff
@@ -1,14 +1,103 @@
...
+    @Test
+    public void testValidateResponsesDetails_shouldReturnEmptyList() {
+        List<FeedbackResponseDetails> responses = new ArrayList<>();
+        int numRecipients = 0;
+
+        assertTrue(feedbackTextQuestionDetails.validateResponsesDetails(responses, numRecipients).isEmpty());
+
+        for (int i = 0; i < 10; i++) {
+            responses.add(new FeedbackTextResponseDetails());
+        }
```
Comment (halfwhole) [inline src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java:87]: Perhaps we won't need a loop here, maybe just add once or twice will do? (The method under test is also straightforward so might prefer simpler solutions)

File: src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java [hunk 2]
```diff
@@ -1,14 +1,103 @@
...
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        feedbackTextQuestionDetails.setRecommendedLength(-1);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        feedbackTextQuestionDetails.setRecommendedLength(-100);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        ______TS("Recommended length is not null and is greater than or equal to 1; returns an empty list");
+        expectedResult.clear();
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java:67]: Ok, thank you for the advice.

File: src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java [hunk 3]
```diff
@@ -1,14 +1,103 @@
...
+        assertTrue(feedbackTextQuestionDetails.validateResponsesDetails(responses, numRecipients).isEmpty());
+
+        for (int i = 0; i < 10; i++) {
+            responses.add(new FeedbackTextResponseDetails());
+        }
+        numRecipients = 10;
+        assertTrue(feedbackTextQuestionDetails.validateResponsesDetails(responses, numRecipients).isEmpty());
+
+        numRecipients = -100;
+        assertTrue(feedbackTextQuestionDetails.validateResponsesDetails(responses, numRecipients).isEmpty());
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java:92]: Ok, thank you for the advice.

File: src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java (hunk 1 above)
Comment (hhdqirui) [reply src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java:87]: Ok, thank you for the advice.

**Comment by hhdqirui** [comment]
Hi @halfwhole, it is all fixed now and is ready for review.

File: src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java [hunk 4]
```diff
@@ -1,14 +1,99 @@
...
+        feedbackTextQuestionDetails.setRecommendedLength(100);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+    }
+
+    @Test
+    public void testValidateResponsesDetails_shouldReturnEmptyList() {
+        List<FeedbackResponseDetails> responses = new ArrayList<>();
+        int numRecipients = 0;
+
+        assertTrue(feedbackTextQuestionDetails.validateResponsesDetails(responses, numRecipients).isEmpty());
```
Comment (halfwhole) [inline src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java:83]: This `numRecipients` here can be substituted too

**Review by halfwhole — COMMENTED** [review]
Sorry for taking several days to get back, LGTM apart from one minor comment

**Review by halfwhole — APPROVED** [review]

**Review by rrtheonlyone — APPROVED** [review]
LGTM

**Pull request merged**